From 459d63c81c87e23899af5eadc8524e4671b757f3 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 25 Dec 2020 17:58:15 +0100
Subject: [PATCH 667/985] evl/sched: do not consider oob stall on enabling
 preemption

Checking the oob stall bit in __evl_enable_preempt() to block the
rescheduling is obsolete. It relates to a nested locking construct
which is long gone, when the evl_spinlock managed the preemption count
and the big lock was still in, i.e.:

lock_irqsave(&ugly_big_lock, flags)  /* stall bit raised */
	evl_spin_lock(&inner_lock);  /* +1 preempt */
	   wake_up_high_prio_thread();
	evl_spin_unlock(&inner_lock); /* -1 preempt == 0, NO schedule because stalled */
unlock_irqrestore(&ugly_big_lock, flags) /* stall bit restored */

This was a way to prevent a rescheduling to take place inadvertently
while holding the big lock.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/sched.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/evl/sched.h b/include/evl/sched.h
index 4d32a03e2..6318a3eb1 100644
--- a/include/evl/sched.h
+++ b/include/evl/sched.h
@@ -333,8 +333,7 @@ static inline void __evl_disable_preempt(void)
 
 static inline void __evl_enable_preempt(void)
 {
-	if (--dovetail_current_state()->preempt_count == 0 &&
-		!oob_irqs_disabled())
+	if (--dovetail_current_state()->preempt_count == 0)
 		evl_schedule();
 }
 
-- 
2.30.2

